lib/sdl/Makefile

```diff
@@ -13,9 +16,22 @@ FREETYPE_SRCS += $(FREETYPE_DIR)/src/raster/raster.c
 FREETYPE_SRCS += $(FREETYPE_DIR)/src/smooth/smooth.c
 FREETYPE_SRCS += $(FREETYPE_DIR)/src/autofit/autofit.c
 FREETYPE_SRCS += $(FREETYPE_DIR)/src/sfnt/sfnt.c
+FREETYPE_OBJS = $(addsuffix .obj, $(basename $(FREETYPE_SRCS)))
+
+$(NXDK_DIR)/lib/libfreetype.lib: $(FREETYPE_OBJS)
 
-SRCS += $(SDL_TTF_SRCS) $(FREETYPE_SRCS)
 CFLAGS += -I$(NXDK_DIR)/lib/sdl \
           -I$(SDL_TTF_DIR) \
           -I$(FREETYPE_DIR)/include \
           -DXBOX
+
+main.exe: $(NXDK_DIR)/lib/libSDL_ttf.lib $(NXDK_DIR)/lib/libfreetype.lib
```

Comment:
What main.exe is this?

Reply:
The `main.exe` of whatever sample you're building at that moment. We're just adding a library dependecy here (only if SDL was requested ofc).

As we move away from the "one big makefile handles everything" approach, I expect samples to list their required libs explicitly in their own Makefile (with proper library directory set in the linker flags etc.) in the future, this is only for the transitioning phase.

Reply:
I'd have expected libraries to be linked like libxboxkrnl.lib (which is linked from the Makefile in the nxdk root).

I didn't even know you could add more dependencies to the same output using this (I thought each main.exe line would create a separate target).

However, suspecting this would add dependencies to application default.xbe, my comment was also meant to hint at the fact that a library Makefile should be independent of the users project.

> this is only for the transitioning phase.

Fair enough. In that case this is fine with me.